pkg/zson/record.go

```diff
@@ -157,6 +173,34 @@ func (r *Record) Strings() ([]string, error) {
 	return ss, nil
 }
 
+// TypeCheck checks that the value coding in Raw is structurally consistent
+// with this value's descriptor.  It does not check that the actual leaf
+// values when parsed are type compatible with the leaf types.
+func (r *Record) TypeCheck() bool {
+	return checkRecord(r.Descriptor.Type, r.Raw)
+}
+
+func checkRecord(typ *zeek.TypeRecord, body zval.Encoding) bool {
+	it := zval.Iter(body)
+	for _, col := range typ.Columns {
+		body, container, err := it.Next()
+		if err != nil {
+			return false
+		}
+		switch v := col.Type.(type) {
+		case *zeek.TypeRecord:
+			return checkRecord(v, body)
+		case *zeek.TypeSet, *zeek.TypeVector:
+			return container
```

Comment:
don't we need to recurse into container here for containers of containers?

Reply:
yes, for vectors.  done.  sets can't have containers inside of them.

Reply:
hmm, I guess we should check for that?